### PR TITLE
aws sdk 3.x compatibility

### PR DESCRIPTION
**Pull request by nicolasricci**
add parameter to specify aws_sdk version
update configuration and metadatabuilder to comply with aws sdk 3.x

### CODE REVIEW

File: DependencyInjection/SonataMediaExtension.php [hunk 1]
```diff
@@ -379,12 +379,34 @@ public function configureFilesystemAdapter(ContainerBuilder $container, array $c
                 ))
             ;
 
+            if($config['filesystem']['s3']['sdk_version'] === 3){
```
Comment (OskarStark) [inline DependencyInjection/SonataMediaExtension.php]: spaces missing after `if` and before `{`

File: DependencyInjection/SonataMediaExtension.php [hunk 2]
```diff
@@ -379,12 +379,34 @@ public function configureFilesystemAdapter(ContainerBuilder $container, array $c
...
-                    'key'    => $config['filesystem']['s3']['accessKey'],
+                    'credentials' => array(
+                        'secret' => $config['filesystem']['s3']['secretKey'],
+                        'key'    => $config['filesystem']['s3']['accessKey'],
+                    ),
+                    'region' => $config['filesystem']['s3']['region'],
+                    'version' => $config['filesystem']['s3']['version'],
+                ))
+            ;
+            }
```
Comment (OskarStark) [inline DependencyInjection/SonataMediaExtension.php]: `} else {` should be on one line

File: Metadata/AmazonMetadataBuilder.php [hunk 3]
```diff
@@ -27,12 +38,12 @@ class AmazonMetadataBuilder implements MetadataBuilderInterface
      * @var string[]
      */
     protected $acl = array(
-        'private'            => CannedAcl::PRIVATE_ACCESS,
-        'public'             => CannedAcl::PUBLIC_READ,
-        'open'               => CannedAcl::PUBLIC_READ_WRITE,
-        'auth_read'          => CannedAcl::AUTHENTICATED_READ,
-        'owner_read'         => CannedAcl::BUCKET_OWNER_READ,
-        'owner_full_control' => CannedAcl::BUCKET_OWNER_FULL_CONTROL,
+        'private' => 'private',
```
Comment (OskarStark) [inline Metadata/AmazonMetadataBuilder.php]: if you add all the constants, you should use them here

**Comment by nicolasricci** [comment]
thanks @OskarStark. I have fixed the constants bit and run the `make cs` to fix indentation issues

**Comment by nicolasricci** [comment]
this PR is fixing this issue #827 btw

File: DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -215,6 +215,8 @@ private function addFilesystemSection(ArrayNodeDefinition $node)
                                     ->end()
                                 ->end()
                                 ->scalarNode('region')->defaultValue('s3.amazonaws.com')->end()
+                                ->scalarNode('version')->defaultValue('latest')->end()
```
Comment (rande) [inline DependencyInjection/Configuration.php]: Can you explain the usage of this settings ? Maybe you should add some note in the documentation.

Comment (nicolasricci) [inline DependencyInjection/Configuration.php]: @rande documentation has been updated

**Comment by OskarStark** [comment]
can you please check travis @nicolasricci

**Comment by OskarStark** [comment]
restarted travis

**Comment by OskarStark** [comment]
strange...

@core23 can you help with the composer timeout issue ?

**Comment by core23** [comment]
No, got the same problems with my PR https://github.com/sonata-project/SonataFormatterBundle/pull/142

**Comment by greg0ire** [comment]
Try getting rid of `minimum-stability` and `prefer-stable` if they are present.

**Comment by OskarStark** [comment]
ping @nicolasricci

**Comment by nicolasricci** [comment]
sorry @OskarStark i did not receive the notifications for some reason. Tests are now failing because the dependencies @Soullivaneuh is working on in PR #990  are still not merged.

**Comment by nicolasricci** [comment]
@OskarStark can this be reviewed for merge now that PR #990 has been merged. Not sure why tests are still failing though...

**Comment by OskarStark** [comment]
did you rebase and force push to trigger travis again?

**Comment by OskarStark** [comment]
@nicolasricci can you please use the default PR header?

**Comment by Soullivaneuh** [comment]
According to the new [Sonata version management](https://github.com/sonata-project/SonataAdminBundle/issues/3053) and next major [release plan](https://github.com/sonata-project/SonataAdminBundle/issues/3731), this project has been refactored regarding branching and versioning.

If you see this message, your PR concerns a patch or a minor release and is not targeting the right branch.

So I'm closing this one, but don't see it as a refusal. If you think your work is still relevant and want to continue, feel free to reopen it on the right branch (e.g. the default one).

Regards.

**Comment by nicolasricci** [comment]
@Soullivaneuh first of all thanks for all the hard work on Sonata project, it was clearly a good idea to have this release plan.
If I got it right I am supposed to resubmit my PR on the 3.x branch?

**Comment by greg0ire** [comment]
>  If I got it right I am supposed to resubmit my PR on the 3.x branch? 

yes

**Comment by nicolasricci** [comment]
👍

**Pull request closed without merging**